crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -65,6 +65,14 @@ pub struct TransactionHandle {
     pub tx_hash: H256,
 }
 
+#[derive(Debug, Clone)]
+pub struct CancelHandle {
+    /// transaction previosly submitted using TransactionSubmitting::submit_transaction()
+    pub submitted_transaction: TransactionHandle,
```

Comment:
This doesn't seem to be used anywhere. Is this for debugging purposes only?

Reply:
Nice catch. A bit of context: we have two different ways of cancelling the submitted transaction:
1. Using the previously submitted transaction hash/bundle_id (Flashbots are using this way)
2. Using the noop transaction (CustomNodes and Eden are using this way)
Seems like an Enum would be a perfect solution for this, however `CancelHandle` is populated inside `Submitter` which abstracts type of strategy, therefore we cant decide on the `CancelHandle` variant inside it, therefore we need to populate both 1 and 2 in order to provide valid `CancelHandle`.

And the response to the question, `submitted_transaction` is not used because currently Flashbots cancellation implementation is missing since its not mandatory for this PR experiment.

Reply:
I see. Thanks for the detailed explanation. Will resolve the other conversation then. 👍